src/dev/precommit_hook/get_files_for_commit.js

```diff
@@ -0,0 +1,40 @@
+import SimpleGit from 'simple-git';
+import { fromNode as fcb } from 'bluebird';
+import { File } from '../file';
+
+/**
+ * Get the files that are staged for commit (excluding deleted files)
+ * as `File` objects that are aware of their commit status.
+ *
+ * @param  {String} repoPath
+ * @return {Promise<Array<File>>}
+ */
+export async function getFilesForCommit(repoPath) {
```

Comment:
__nit:__ let's either remove `repoPath` argument or pass `process.cwd()` as argument when it's called from `run_precommit_hook`.